lib/services/address/index.js

```diff
@@ -397,6 +407,13 @@ AddressService.prototype.updateMempoolIndex = function(tx, add, callback) {
       value: inputValue
     });
 
+    var addressIndexKey = encoding.encodeMempoolAddressIndexKey(inputHashBuffer, inputHashType);
```

Comment:
We may need to have these be separate indexes. If there are two transactions in the mempool for an address where one is receiving to an address and the other spending from an address, and one of these transactions is removed from the mempool (was included in a block). The index will be removed and then query for the still in mempool transaction would be skipped.